frontend/server/src/Controllers/Problem.php

```diff
@@ -3488,11 +3521,14 @@ public static function getProblemDetailsForSmarty(
 
         // Get problem details from API
         $details = self::getProblemDetails(
-            $r,
+            $r->identity,
             $problem,
             $problemset,
             strval($r['lang']),
-            /*showSolvers=*/true
+            /*showSolvers=*/true,
+            $r['auth_token'],
+            boolval($r['prevent_problemset_open']) === true,
```

Comment:
```suggestion
            boolval($r['prevent_problemset_open']),
```

el `=== true` es implícito porque `boolval()` sólo puede regresar `true` o `false`.

Reply:
Listo, se quita.